### PR TITLE
text implementation and flare3d init

### PR DESCRIPTION
**Pull request by 0123456789-jpg**
Not done yet. Just making it compatible with new crate structure.

### CODE REVIEW

File: core/src/util/formatting.rs
Comment (0123456789-jpg) [inline core/src/util/formatting.rs:1]: Should this and StringIdentifiable be in rimecraft_primitives? @JieningYu

Comment (JieningYu) [reply core/src/util/formatting.rs:1]: > Should this and StringIdentifiable be in rimecraft_primitives? @JieningYu

you won't need it with `serde`

**Comment by JieningYu** [comment]
wow erased

**Comment by JieningYu** [comment]
wow rendering stuff

File: flare3d/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+[badges]
+maintenance = { status = "actively-developed" }
+
+[dependencies]
+# Wgpu
+winit = "0.26.0"
+wgpu = "0.12.0"
+env_logger = "0.9"
+log = "0.4"
```
Comment (JieningYu) [inline flare3d/Cargo.toml:19]: use `tracing` crate if possible

Comment (KrLite) [reply flare3d/Cargo.toml]: ok

**Comment by 0123456789-jpg** [comment]
hello to matrix

**Comment by JieningYu** [comment]
habixia

**Comment by 0123456789-jpg** [comment]
> habixia

哈比上

**Review by JieningYu — APPROVED** [review]

**Comment by KrLite** [comment]
Great

**Comment by JieningYu** [comment]
p0g

**Comment by 0123456789-jpg** [comment]
WoW

File: core/src/text/tests.rs [hunk 2]
```diff
@@ -49,14 +44,15 @@ fn ser_de() {
...
     );
     text.push(Text::new(
-        super::content::Content::Translatable(Translatable::new(
-            std::borrow::Cow::Borrowed("item.rimecraft.gold_ingot"),
-            None,
-            vec![TranslatableArg::Display(Arc::new(
+        crate::text::content::Content::Empty,
+        vec![Text::new(
+            super::content::Content::Literal(std::borrow::Cow::Borrowed(
                 "玩Blue Archive的人素质都很差",
```
Comment (JieningYu) [inline core/src/text/tests.rs:50]: 同意了 狠狠同意了

Comment (KrLite) [reply core/src/text/tests.rs:50]: qwq

**Pull request merged**